common/src/main/java/com/microsoft/identity/common/internal/cache/MSALOAuth2TokenCache.java

```diff
@@ -12,22 +13,53 @@
 
 public class MSALOAuth2TokenCache extends OAuth2TokenCache implements IShareSingleSignOnState {
 
+    // SharedPreferences used to store tokens
+    private final SharedPreferencesFileManager mAccessTokenSharedPreferences;
+    private final SharedPreferencesFileManager mRefreshTokenSharedPreferences;
+
+    // The names of the SharedPreferences files on disk
+    private static final String sAccessTokenSharedPreferences = "com.microsoft.identity.client.token";
+    private static final String sRefreshTokenSharedPreferences = "com.microsoft.identity.client.refreshToken";
+
     public MSALOAuth2TokenCache(Context context) {
         super(context);
+        mAccessTokenSharedPreferences = new SharedPreferencesFileManager(mContext, sAccessTokenSharedPreferences);
+        mRefreshTokenSharedPreferences = new SharedPreferencesFileManager(mContext, sRefreshTokenSharedPreferences);
     }
 
     @Override
     public void setSingleSignOnState(Account account, RefreshToken refreshToken) {
-
+        // TODO implement
     }
 
     @Override
     public RefreshToken getSingleSignOnState(Account account) {
+        // TODO implement
         return null;
     }
 
     @Override
     public void saveTokens(OAuth2Strategy oAuth2Strategy, AuthorizationRequest request, TokenResponse response) {
+        saveAccessToken(oAuth2Strategy, request, response);
+        saveRefreshToken(oAuth2Strategy, request, response);
+    }
+
+    private void saveAccessToken(OAuth2Strategy oAuth2Strategy, AuthorizationRequest request, TokenResponse response) {
+        // TODO delete old tokens? Does this really make sense to do here?
```

Comment:
There's [some logic in the existing MSAL code to do this](https://github.com/AzureAD/microsoft-authentication-library-for-android/blob/dev/msal/src/cache/java/com/microsoft/identity/client/TokenCache.java#L76-L83): I guess that needs to get ported over?